openedx/features/enterprise_support/tests/test_api.py

```diff
@@ -90,8 +94,41 @@ def setUpTestData(cls):
             email='ent_worker@example.com',
             password='password123',
         )
+        cls.enterprise_customer = EnterpriseCustomerFactory()
+        cls.pending_enterprise_customer_user = PendingEnterpriseCustomerUserFactory()
         super().setUpTestData()
 
+
+    @mock.patch('openedx.features.enterprise_support.api.create_jwt_for_user')
+    def test_list_learners_filtered(self, mock_jwt_builder):
+        """
+        Test that the list learners endpoint can be filtered by user details
+        """
+        user = UserFactory(email='foobar@example.com')
+        UserProfileFactory.create(
+            user=user,
+            defaults={'name': 'Test Enterprise User'},
+        )
+        group = EnterpriseGroupFactory(
+            enterprise_customer=self.enterprise_customer,
+        )
+        pending_user = PendingEnterpriseCustomerUserFactory(
+            user_email="foobar@example.com",
+            enterprise_customer=self.enterprise_customer,
+        )
+
+        pending_user_query_string = f'?user_query={pending_user.user_email}'
+        # approach 1
+        api_client = self._assert_api_client_with_user(EnterpriseApiClient, mock_jwt_builder)
+        response = api_client.get_learners(group.uuid, pending_user_query_string)
+        assert response.status_code == 200
```

Comment:
error msg: 

`requests.exceptions.ConnectionError: HTTPConnectionPool(host='enterprise.example.com', port=80): Max retries exceeded with url: /enterprise/api/v1/enterprise-group/72e52eee-9ff0-4a2a-a931-cb2f91c87782/learners/?user_query=foobar@example.com (Caused by NameResolutionError("<urllib3.connection.HTTPConnection object at 0x75ce02c59b10>: Failed to resolve 'enterprise.example.com' ([Errno -2] Name or service not known)"))`

Reply:
I think you need to have this test mock the real client object at `EnterpriseApiClient.return_value.client` so your test doesn't actually try to call the fake test server that's set up for unit tests. You'd do that with maybe a decorator to mock the `EnterpriseApiClient` like this:
```python
@mock.patch('openedx.features.enterprise_support.EnterpriseApiClient)
@mock.patch('openedx.features.enterprise_support.api.create_jwt_for_user')
def test_list_learners_filtered(self, mock_jwt_builder, mock_enterprise_client):
    mock_rest_client = mock_enterprise_client.return_value.client # mocks out the call to self.client.get() in your get_learners() method
```